modules/signing_worker/manifests/system_user.pp

```diff
@@ -0,0 +1,66 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+class signing_worker::system_user (
+    String $user = 'cltbld',
+) {
+    case $::operatingsystem {
+        'Darwin': {
+            $password     = lookup("${user}_user.password")
+            $salt         = lookup("${user}_user.salt")
+            $iterations   = lookup("${user}_user.iterations")
+            $kcpassword   = lookup("${user}_user.kcpassword")
+
+            # Create the cltbld user
+            users::single_user { $user:
+                # Bug 1122875 - cltld needs to be in this group for debug tests
+                password   => $password,
+                salt       => $salt,
+                iterations => $iterations,
+            }
+
+            # Monkey patching directoryservice.rb in order to create users also breaks group merging
+            # So we directly add the user to the group(s)
+            exec { "${user}_developer_group":
+                command => "/usr/bin/dscl . -append /Groups/_developer GroupMembership ${user}",
+                unless  => "/usr/bin/groups ${user} | /usr/bin/grep -q -w _developer",
+                require => User[$user],
+            }
+
+            # Set user to autologin
+            class { 'macos_utils::autologin_user':
+                user       => $user,
+                kcpassword => $kcpassword,
+            }
+
+            # Enable DevToolsSecurity
+            include macos_utils::enable_dev_tools_security
+
+            macos_utils::clean_appstate { $user:
+                user  => $user,
+                group => 'staff',
+            }
+
+            mercurial::hgrc { "/Users/${user}/.hgrc":
+                user    => $user,
+                group   => 'staff',
+                require => User[$user],
+            }
+
+            sudo::custom { "allow_${user}_reboot":
+                user    => $user,
+                command => '/sbin/reboot',
+            }
+
+            # Consider removing this if/when 'sudo su' is no longer used.
+            sudo::custom { "allow_${user}_all":
+                user    => $user,
+                command => 'ALL',
+            }
```

Comment:
Hmm, which also means that the lines we're getting overwritten on the test nodes (see slack, and comments above) also need reworking. How should we best combine rules that say 'allow reboot as root, and access to the following other users' into one line?

Reply:
If need be, we could rename the scriptworker prod user to `cltsign`, which a) matches what we had on the signing servers and linux signing scriptworkers, and b) doesn't have a conflicting line in sudoers. Only mentioning this in case it helps work around the problem.